### PR TITLE
feat: Adds Python script and requirements to harvest the latest versions (acceptance tests)

### PR DESCRIPTION
**Pull request by maximearmstrong**
**Summary:**

This PR adds the Python script and requirements to harvest the datasets latest versions URLs from the Mobility Database/archives. The file are under `acceptance/tests/`.

**Expected behavior:** 

Same as before. This PR changes nothing to the behaviour of the canonical GTFS validator.

Please make sure these boxes are checked before submitting your pull request - thanks!

- [x] Run the unit tests with `gradle test` to make sure you didn't break anything
- [x] Format the title like "feat: [new feature short description]". Title must follow the Conventional Commit Specification(https://www.conventionalcommits.org/en/v1.0.0/).
- [x] Linked all relevant issues
- [ ] ~Include screenshot(s) showing how this pull request works and fixes the issue(s)~

### CODE REVIEW

**Review by lionel-nj — CHANGES_REQUESTED** [review]
Thanks for working on that @maximearmstrong! I would suggest:
- creating a Python package via IntelliJ named `scripts` (so that the Python interpreter is automatically configured when opening the project for IntelliJ IDE users)
  Capture d’écran, le 2021-07-14 à 10 00 59

- placing these files inside a directory named `mobility-database-harvester` or `mobility-database-harvest` (it might increase clarity since `acceptance/tests` may suggest that the script you provided is a test suite).
- passing the content of the credentials file used for the authentication to google storage services: locally the path to the said file is provided as an environment variable. However, in the CI process, we would be using the content of the said file instead of its path.

File: acceptance/tests/harvest_latest_versions.py [hunk 1]
```diff
@@ -0,0 +1,175 @@
+import argparse
```
[the remark this answers is not in the document]
Comment (lionel-nj) [reply acceptance/tests/harvest_latest_versions.py:1]: The header should mention the copyright but also the license our code base uses. I copy-pasted the one we used in the Java modules.

File: acceptance/tests/harvest_latest_versions.py [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+    for source in sources:
+        source_id = source[MAINSNAK][DATAVALUE][VALUE][ID]
+        source_data = get_entity_data(source_id)
+        source_archives_id = source_data[CLAIMS][ARCHIVES_ID_PROPERTY][0][MAINSNAK][DATAVALUE][VALUE]
+        archives_ids.append(source_archives_id)
+
+    return harvesting_date, archives_ids
+
+
+def harvest_latest_versions(archives_ids):
```
Comment (lionel-nj) [inline acceptance/tests/harvest_latest_versions.py:98]:
```suggestion
def harvest_latest_versions(archives_ids, credentials):
```

File: acceptance/tests/harvest_latest_versions.py [hunk 3]
```diff
@@ -0,0 +1,175 @@
...
+    client = storage.Client()
```
Comment (lionel-nj) [inline acceptance/tests/harvest_latest_versions.py:99]:
```suggestion
    client = storage.Client.from_service_account_info(credentials=credentials)
```

Comment (lionel-nj) [reply acceptance/tests/harvest_latest_versions.py:99]: Where `credentials` is the stringified content of the credentials (json file) used to access google storage services.

File: acceptance/tests/harvest_latest_versions.py [hunk 4]
```diff
@@ -0,0 +1,175 @@
...
+        default="./data/",
+        help="Data path.",
+    )
+    parser.add_argument(
+        "-c",
+        "--catalog",
+        action="store",
+        choices=["GTFS", "GBFS"],
+        help="Catalog to use for the harvesting. Choices: 'GTFS', 'GBFS'",
+    )
```
Comment (lionel-nj) [inline acceptance/tests/harvest_latest_versions.py:146]:
```suggestion
    )
        parser.add_argument(
        "-x",
        "--credentials",
        action="store",
        help="Grant access to google storage solutions."
    )
```

File: acceptance/tests/harvest_latest_versions.py [hunk 5]
```diff
@@ -0,0 +1,175 @@
...
+        help="Name of the latest versions file. If the file exists, it will be overwritten.",
+    )
+    parser.add_argument(
+        "-d",
+        "--data-path",
+        action="store",
+        default="./data/",
+        help="Data path.",
+    )
+    parser.add_argument(
```
Comment (lionel-nj) [inline acceptance/tests/harvest_latest_versions.py:140]: Should we remove this CLI parameter, since we will only harvest `GTFS` catalogs here.

Comment (maximearmstrong) [reply acceptance/tests/harvest_latest_versions.py:140]: Won't we harvest GBFS datasets in the future? I can remove it if not

Comment (lionel-nj) [reply acceptance/tests/harvest_latest_versions.py:140]: Probably, but no in this repository.

File: acceptance/tests/harvest_latest_versions.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (maximearmstrong) [reply acceptance/tests/harvest_latest_versions.py:1]: Thanks! I did not find it yesterday

File: acceptance/tests/harvest_latest_versions.py (hunk 2 above)
Comment (maximearmstrong) [reply acceptance/tests/harvest_latest_versions.py:98]: I will change this, but never do a `json.loads()` in a signature please 😅

File: acceptance/tests/harvest_latest_versions.py (hunk 3 above)
Comment (maximearmstrong) [reply acceptance/tests/harvest_latest_versions.py:99]: Perfect

File: acceptance/tests/harvest_latest_versions.py (hunk 5 above)
Comment (maximearmstrong) [reply acceptance/tests/harvest_latest_versions.py:140]: Removed!

**Comment by lionel-nj** [comment]
As discussed, I will provide some modifications in order to avoid passing the JSON credentials file as a Github secret.

**Review by lionel-nj — APPROVED** [review]
Thanks @maximearmstrong

**Pull request merged**